discord/utils.py

```diff
@@ -334,6 +339,9 @@ def oauth_url(
         url += '&response_type=code&' + urlencode({'redirect_uri': redirect_uri})
     if disable_guild_select:
         url += '&disable_guild_select=true'
+    if state is not MISSING:
+        encoded = urlencode({'state': state})
```

Comment:
Looks like the import would have to be moved. I didn't realise it was a function local import.

Reply:
yeah lol wait, i will fix that asap as i just simple commiting the sugesstion